src/itoolkit/itoolkit.py

```diff
@@ -386,7 +386,7 @@ def __init__(self, ikey, iname, iopt={}):
         super(iPgm, self).__init__(iopt, opts)
         self.pcnt = 0
 
-    def addParm(self, obj): # noqa N802
+    def addParm(self, obj, iopt={}): # noqa N802
```

Comment:
I'd really like to make this package more Pythonic, so adding more *i* foo stuff is not helping :wink: 

I'd prefer to see something like 
```suggestion
    def addParm(self, obj, by_value=False): # noqa N802
```

Reply:
I fully agree that something more Pythonic would be better. I actually mulled over precisely the change you're proposing, but in the end I went the way I did because

(1) I have a very limited understanding of XMLSERVICE and don't know what other options `iParm` might be able to take. I specifically wanted to expose the entire `iParm` interface from `addParm`.

(2) If we really want to be Pythonic, then maybe the very notion of `addParm` is already too clumsy. Maybe we need a whole wrapper ("itoolkit for humans") or an additional module that provides a completely revamped API.

(3) Given the above two points, I felt I might as well try to minimize the diff for this commit, as well as preserve the existing overall style.

But if you think we should start chipping away, in this module, I'm game.

Reply:
AFAICT, it only supports `io` and `by`: http://yips.idevcloud.com/wiki/index.php/XMLService/XMLSERVICEQuick

```
      * pgm parameters
      * <parm [io='in|out|both|omit'
      *        by='val|ref'
      *        ]>values (see <ds> and <data>)</parm>
      * ---
      * parm        - parm name (tag)
      *  values     - (see ds or data)
      * options
      *  io
      *     in      - input only
      *     out     - output only
      *     both    - input/output only (default)
      *     omit    - omit (1.2.3)
      *  by
      *     ref     - pass by reference (default)
      *     val     - pass by value (1.9.9.3+)
```
I really would like to start making this entire package more Pythonic and agree we really need a rethink of the API. The current design is really designed around exposing XMLSERVICE to Python, whereas I think a better approach would be to provide a sensible set of Pythonic APIs and then mapping that to whatever nonsense XMLSERVICE needs.

I think in the meantime, since `addParam` isn't even Pythonic, we can be fine with just adding a direct options through. We can always add something like `add_parameter` which has more Pythonic ways of doing things and deprecate the former.